application/models/QuestionAttribute.php

```diff
@@ -578,4 +589,39 @@ public static function getOwnQuestionAttributesViaPlugin()
 
         return (array) $result->get('questionAttributes');
     }
+
+    /**
+     * Apply XSS filter to question attribute value unless 'xssfilter' property is false.
+     * @param string $attribute the name of the attribute to be validated.
+	 * @param array<mixed> $params additional parameters passed with rule when being executed.
+     * @return void
+     */
+    public function filterXss($attribute, $params)
+    {
+        $question = Question::model()->find("qid=:qid", ['qid' => $this->qid]);
+        if (empty($question)) {
+            return;
+        }
+        $questionAttributeFetcher = new \LimeSurvey\Models\Services\QuestionAttributeFetcher();
+        $questionAttributeFetcher->setQuestion($question);
+        $questionAttributeDefinitions = $questionAttributeFetcher->fetch();
+
+        // The value will be filtered unless the attribute definition has the "xssfilter" property set to false
+        $shouldFilter = true;
+        if (isset($questionAttributeDefinitions[$this->attribute])) {
+            $questionAttributeDefinition = $questionAttributeDefinitions[$this->attribute];
+            if (array_key_exists("xssfilter", $questionAttributeDefinition) && $questionAttributeDefinition['xssfilter'] == false) {
+                $shouldFilter = false;
+            }
+        }
+
+        if (!$shouldFilter) {
```

Comment:
And then directly `!$questionAttributeDefinition['xssfilter']` or `$questionAttributeDefinition['xssfilter'] === false` : less code, less test, less variable creation : good for speed and comprehension (with getDefaultSettings)